track/admin.py

```diff
@@ -13,6 +15,35 @@ class OrderAdmin(admin.ModelAdmin):
     change_list_template = "track/check_orders.html"
 
     list_display = ["record_id", "order_number", "tracking_nrs", "return_tracking_nrs", "tube_serials", "order_status", "ship_date"]
+    actions = ['check_orders_status']
+
+    def get_actions(self, request):
+        actions = super().get_actions(request)
+        return actions
+
+    def check_orders_status(self, request, queryset):
+        if not queryset:
+            self.message_user(request, "No orders selected to check status.", messages.WARNING)
+            return
+            
+        try:
+            order_numbers = list(queryset.values_list('order_number', flat=True))
+            queryset.update(order_status=Order.INITIATED)
+            
+            # Check shipping info for selected orders only
+            tracking_info = gbf.get_order_confirmations(order_numbers)
+            if tracking_info:
+                orders._update_orders_with_shipping_info(tracking_info)
+                message = f"Successfully checked shipping status for orders: {', '.join(order_numbers)}"
+                self.message_user(request, message, messages.SUCCESS)
+            else:
+                self.message_user(request, "No shipping information available for the selected orders.", messages.INFO)
+                
+        except Exception as e:
+            logger.error(f"Error checking order status: {str(e)}")
+            self.message_user(request, f"Error checking order status: {str(e)}", messages.ERROR)
+    
+    check_orders_status.short_description = "Check shipping status for selected orders"
```

Comment:
you'll need to start a log with the log manager at the beginning and complete it at the end
```
log_manager.start_confirmation_log()
...
log_manager.complete_log()
```